src/common/session/session-entity.component-test.js

```diff
@@ -235,6 +235,18 @@ describe('reset()', () => {
   })
 })
 
+test('isNew is true after the session resets by timers', async () => {
+  const sessionInstance = new SessionEntity({ agentIdentifier, key, storage })
+  expect(sessionInstance.isNew).toEqual(true)
+  sessionInstance.reset() // imitate what expireMs and inactiveMs does on activation
+  expect(sessionInstance.isNew).toEqual(true)
+})
+test('isNew is true if reset happens on initialization after time outs off-view', () => {
+  storage.set(`${PREFIX}_${key}`, { ...model, value, expiresAt: Date.now() - 1, inactiveAt: Date.now() - 1, updatedAt: Date.now() })
+  const sessionInstance = new SessionEntity({ agentIdentifier, key, storage })
+  expect(sessionInstance.isNew).toEqual(true)
+})
+
```

Comment:
```suggestion
describe('isNew', () => {
  test('isNew is true after the session resets by timers', async () => {
    const sessionInstance = new SessionEntity({ agentIdentifier, key, storage })
    expect(sessionInstance.isNew).toEqual(true)
    sessionInstance.reset() // imitate what expireMs and inactiveMs does on activation
    expect(sessionInstance.isNew).toEqual(true)
  })
  test('isNew is true if reset happens on initialization after time outs off-view | expiresAt', () => {
    storage.set(`${PREFIX}_${key}`, { ...model, value, expiresAt: Date.now() - 1, inactiveAt: Date.now() + 100000, updatedAt: Date.now() })
    const sessionInstance = new SessionEntity({ agentIdentifier, key, storage })
    expect(sessionInstance.isNew).toEqual(true)
  })
  test('isNew is true if reset happens on initialization after time outs off-view | inactiveAt', () => {
    storage.set(`${PREFIX}_${key}`, { ...model, value, expiresAt: Date.now() + 100000, inactiveAt: Date.now() - 1, updatedAt: Date.now() })
    const sessionInstance = new SessionEntity({ agentIdentifier, key, storage })
    expect(sessionInstance.isNew).toEqual(true)
  })
  test('isNew is true if reset happens on initialization after time outs off-view | invalid data', () => {
    const storage = new LocalMemory({ [`${PREFIX}_${key}`]: { invalid_fields: true } })
    const sessionInstance = new SessionEntity({ agentIdentifier, key, storage })
    expect(sessionInstance.isNew).toEqual(true)
  })
})
```

Reply:
nice work on this.  the above just splits the test apart to ensure that both expires and inactive timestamps (and invalid data in general) still lead to a positive isNew value